### PR TITLE
[4.0] Subform Cleanup

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
In the PR https://github.com/joomla/joomla-cms/pull/33200 (which was merged), I renamed the plugin from **Subfields** to **Subform**. However, there are still few instances of **Subfields** which should be updated to **Subform**. This PR just makes that final clean up.

### Testing Instructions
1. In update package for the original PR https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33200/downloads/42161/Joomla_4.0.0-beta8-dev+pr.33200-Development-Update_Package.zip . You can go to System -> Update -> Joomla to update that package
2. Access to Content -> Fields, create a new Subform custom field
3. Before patch, in the Fields setting of that Subform custom field, you can choose the same field multiple times and the system still save it
4. After patch, if you choose same field multiple times, only one will be saved
subform-fields

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4dc9a40bc57748f6a74679553da0dadc28de2022<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33262">issues.joomla.org/tracker/joomla-cms/33262</a>.</sub>

File: administrator/components/com_fields/src/Table/FieldTable.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public function bind($src, $ignore = '')
 		if (isset($src['fieldparams']) && is_array($src['fieldparams']))
 		{
 			// Make sure $registry->options contains no duplicates when the field type is subfields
```
Comment (Quy) [inline administrator/components/com_fields/src/Table/FieldTable.php]:
```suggestion
			// Make sure $registry->options contains no duplicates when the field type is subform
```

**Comment by eddiekonczal** [comment]
By "Subform custom field" do you mean a "Subfields" custom field?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33262">issues.joomla.org/tracker/joomla-cms/33262</a>.</sub>

**Comment by eddiekonczal** [comment]
Also, I'm getting 404 not found for:

https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33200/downloads/42161/Joomla_4.0.0-beta8-dev+pr.33200-Development-Update_Package.zip<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33262">issues.joomla.org/tracker/joomla-cms/33262</a>.</sub>

**Comment by Quy** [comment]
Please use the download link at the bottom of the page.

download

**Comment by eddiekonczal** [comment]
What is the URL of the page with the Download link?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33262">issues.joomla.org/tracker/joomla-cms/33262</a>.</sub>

**Comment by Quy** [comment]
https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33262/downloads/42332/

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 4af60a906ab1222a64c87133c99a9c00a16ef120<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33262">issues.joomla.org/tracker/joomla-cms/33262</a>.</sub>

**Comment by richard67** [comment]
I've restored @Quy 's test result in the issue tracker so it's counted properly. The commit which had invalidated it was just a change in a comment.

**Comment by richard67** [comment]
RTC again (somehow the bot removed it)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33262">issues.joomla.org/tracker/joomla-cms/33262</a>.</sub>

**Comment by PhilETaylor** [comment]
This broke :) https://github.com/joomla/joomla-cms/issues/33326

**Comment by eddiekonczal** [comment]
> 
> 
> https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33262/downloads/42332/

When I go to the above URL and try to download the update package, I get:

**Not Found

The requested URL was not found on this server.**

Download link:

https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33262/downloads/42332/Joomla_4.0.0-beta8-dev+pr.33262-Development-Update_Package.zip

**Comment by joomdonation** [comment]
Thanks @eddiekonczal. This PR was merged, so we do not need to test it anymore.

**Comment by PhilETaylor** [comment]
You can of course use the Nightly build if you want to see this now its merged https://developer.joomla.org/nightlies/Joomla_4.0.0-beta8-dev-Development-Full_Package.zip

**Pull request merged**